src/easylink/devtools/implementation_creator.py

```diff
@@ -91,22 +95,85 @@ def _extract_step_name(script_path: Path) -> str:
         .. code-block:: python
             # STEP_NAME: blocking
         """
-        ...
+        step_name = _extract_metadata("STEP_NAME", script_path)
+        if len(step_name) == 0:
+            raise ValueError(
+                f"Could not find a step name in {script_path}. "
+                "Please ensure the script contains a comment of the form '# STEP_NAME: <name>'"
+            )
+        return step_name[0]
 
 
 class PythonRecipe:
     """A singularity recipe generator specific to implementations written in Python."""
 
-    def __init__(self, script_path: Path) -> None:
+    BASE_IMAGE = (
+        "python@sha256:1c26c25390307b64e8ff73e7edf34b4fbeac59d41da41c08da28dc316a721899"
+    )
+
+    def __init__(self, script_path: Path, requirements: str) -> None:
         self.script_path = script_path
-        self.implementation_name = script_path.name
+        self.requirements = requirements
+        self.text: str | None = None
 
     def build(self) -> None:
         """Builds the recipe for the container."""
-        logger.info(f"Building recipe for '{self.implementation_name}'")
-        ...
+        logger.info(f"Building recipe for '{self.script_path.stem}'")
+
+        script_name = self.script_path.name
+        self.text = f"""
+Bootstrap: docker
+From: {self.BASE_IMAGE}
+
+%files
+    ./{script_name} /{script_name}
+
+%post
+    # Create directories
+    mkdir -p /input_data
+    mkdir -p /extra_implementation_specific_input_data
+    mkdir -p /results
+    mkdir -p /diagnostics
+
+    # Install Python packages with specific versions
+    pip install {self.requirements}
+
+%environment
+    export LC_ALL=C
+
+%runscript
+    python /{script_name} '$@'"""
 
     def write(self) -> None:
         """Writes the recipe to disk."""
-        logger.info(f"Writing recipe for '{self.implementation_name}' to disk.")
-        ...
+        logger.info(f"Writing recipe for '{self.script_path.stem}' to disk.")
+        recipe_path = self.script_path.with_suffix(".def")
+        if not self.text:
+            raise ValueError("No recipe text to build.")
+        if recipe_path.exists():
+            logger.warning(f"Recipe file {recipe_path} already exists. Overwriting it.")
+        recipe_path.write_text(self.text)
+        if not recipe_path.exists():
```

Comment:
Should we be concerned about file-system latency here?

Reply:
I'm not terribly concerned b/c Path.write_text() should open the file, write, and then close. I did read some stuff online about how flushing python's buffer can help here, so I went ahead and added that (can't say I totally understand why that helps though).

We can also flush the os buffer using os.fsync, but that seems scary unless we find we actually have problems here.